components/Searchbar.js

```diff
@@ -0,0 +1,151 @@
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useRef, useState, useEffect } from 'react';
+
+export default function Searchbar({ books }) {
+	// Hooks for search bar
+	const searchRef = useRef(null);
+
+	// router
+	const router = useRouter();
+		
+	/**
+	 * @param {HTMLElement} child 
+	 * @param {HTMLElement} parent 
+	 */
+	/*
+	const isDeepChildOf = (child, parent) => {
+		console.log(child.parentNode, parent);
+		if (child == parent) {
+			return true;
+		} else if (child === document.body || child === document.documentElement || document.head || window) {
+			return false;
+		} else {
+			return isDeepChildOf(child.parentNode, parent);
+		}
+	}
+	*/
+
+	useEffect(() => {
```

Comment:
put a comment above this section explaining why it's here and/or what it accomplishes.

Reply:
I know it's hard, but you have to use your imagination to pretend you are reading this code for the first time.

Something like: "Create event listeners to activate search using '`/`' and '`Ctrl + K`'"